DataFormats/METReco/interface/BeamHaloSummary.h

```diff
@@ -70,6 +73,12 @@ namespace reco {
     std::vector<int>& GetGlobaliPhiSuspects() { return GlobaliPhiSuspects ;}
     const std::vector<int>& GetGlobaliPhiSuspects() const { return GlobaliPhiSuspects ;}
 
+    std::vector<std::vector<std::pair<char, CaloTowerDetId> > >& GetProblematicStrips() { return ProblematicStrips ;}
```

Comment:
Vectors of vectors are not good for persisted objects, we need to review these on a case-by-case basis.
You are saving a vector of "strip", where a "strip" is  a   vector of "CaloTowerDetId with a value" in this explicit vector-of-vectors form. A more reasonable solution would be to have a "CaloTowerIdsWithQuality" or a similar name that contains one "strip" and save a vector of these objects here in the BeamHaloSummary.
@Dr15Jones is it going to make things better from the I/O perspective?

I suggest to use uint8_t instead of a char
 (the rest of the code uses char for a byte-sized value though, the old parts could probably stay unchanged).